src/file_pipe_log/format.rs

```diff
@@ -155,6 +156,36 @@ impl LogFileFormat {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct LogFileContext {
+    pub id: FileId,
+    pub version: Version,
+}
+
+impl LogFileContext {
+    pub fn new(file_id: FileId, version: Version) -> Self {
+        Self {
+            id: file_id,
+            version,
+        }
+    }
+
+    #[cfg(test)]
+    pub fn dummy(queue: LogQueue) -> Self {
+        Self {
+            id: FileId::dummy(queue),
+            version: Version::default(),
+        }
+    }
+
+    pub fn get_signature(&self) -> u32 {
+        let high = (self.id.seq as u64 >> 32) as u32;
+        let low = self.id.seq as u32;
+        let flag = self.id.queue as u32;
+        high ^ low ^ flag
```

Comment:
Why are you still doing this? It's wrong to simply xor different fields together. There's absolutely no guarantee that it will yield different values for different files.

As I said, here you should use low bits (which is different for consecutive files less than u32::MAX), and add comment to explain why high and flag is not needed.